docs/docs/reconciliation.md

```diff
@@ -1,49 +1,44 @@
 
 ## Motivation
 
-Generating the minimum number of operations to transform one tree into another is a complex and well-studied problem. The [state of the art algorithms](http://grfia.dlsi.ua.es/ml/algorithms/references/editsurvey_bille.pdf) have a complexity in the order of O(n<sup>3</sup>) where n is the number of nodes in the tree.
+Generating the minimum number of operations to transform one tree into another is a complex and well-studied problem. The [state of the art algorithms](http://grfia.dlsi.ua.es/ml/algorithms/references/editsurvey_bille.pdf) have a complexity in the order of O(n<sup>3</sup>) where n is the number of elements in the tree.
 
-This means that displaying 1000 nodes would require in the order of one billion comparisons. This is far too expensive for our use case. To put this number in perspective, CPUs nowadays execute roughly 3 billion instructions per second. So even with the most performant implementation, we wouldn't be able to compute that diff in less than a second.
+This means that displaying 1000 elements would require in the order of one billion comparisons. This is far too expensive for our use case. To put this number in perspective, CPUs nowadays execute roughly 3 billion instructions per second. So even with the most performant implementation, we wouldn't be able to compute that diff in less than a second.
 
 Since an optimal algorithm is not tractable, we implement a non-optimal O(n) algorithm using heuristics based on two assumptions:
 
-1. Two components of the same class will generate similar trees and two components of different classes will generate different trees.
+1. Two components of the same type will generate similar trees and two components of different types will generate different trees.
 2. It is possible to provide a unique key for elements that is stable across different renders.
 
-In practice, these assumptions are ridiculously fast for almost all practical use cases.
-
+In practice, these assumptions are valid for almost all practical use cases.
 
 ## Pair-wise diff
 
-In order to do a tree diff, we first need to be able to diff two nodes. There are three different cases being handled.
-
+In order to do a tree diff, we first need to be able to diff two elements. There are three different cases being handled.
 
-### Different Node Types
+### Different Element Types
 
-If the node type is different, React is going to treat them as two different sub-trees, throw away the first one and build/insert the second one.
+If the element type is different, React is going to treat them as two different sub-trees, throw away the first one and build/insert the second one.
 
 ```xml
 renderA: <div />
 renderB: <span />
-=> [removeNode <div />], [insertNode <span />]
+=> [removeElement <div />], [insertElement <span />]
```

Comment:
(I don't have a strong opinion about this. If you're happy with notation keep it, but please make operation names unambiguously refer to DOM nodes)